pkg/apis/intelligence/v1alpha1/types.go

```diff
@@ -31,19 +34,30 @@ type NetworkPolicyRecommendation struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Type                string      `json:"jobType,omitempty"`
-	Limit               int         `json:"limit,omitempty"`
-	PolicyType          string      `json:"policyType,omitempty"`
-	StartTime           metav1.Time `json:"startTime,omitempty"`
-	EndTime             metav1.Time `json:"endTime,omitempty"`
-	NSAllowList         []string    `json:"nsAllowList,omitempty"`
-	ExcludeLabels       bool        `json:"excludeLabels,omitempty"`
-	ToServices          bool        `json:"toServices,omitempty"`
-	ExecutorInstances   int         `json:"executorInstances,omitempty"`
-	DriverCoreRequest   string      `json:"driverCoreRequest,omitempty"`
-	DriverMemory        string      `json:"driverMemory,omitempty"`
-	ExecutorCoreRequest string      `json:"executorCoreRequest,omitempty"`
-	ExecutorMemory      string      `json:"executorMemory,omitempty"`
+	Type                string                            `json:"jobType,omitempty"`
+	Limit               int                               `json:"limit,omitempty"`
+	PolicyType          string                            `json:"policyType,omitempty"`
+	IntervalStart       metav1.Time                       `json:"intervalStart,omitempty"`
+	IntervalEnd         metav1.Time                       `json:"intervalEnd,omitempty"`
+	NSAllowList         []string                          `json:"nsAllowList,omitempty"`
+	ExcludeLabels       bool                              `json:"excludeLabels,omitempty"`
+	ToServices          bool                              `json:"toServices,omitempty"`
+	ExecutorInstances   int                               `json:"executorInstances,omitempty"`
+	DriverCoreRequest   string                            `json:"driverCoreRequest,omitempty"`
+	DriverMemory        string                            `json:"driverMemory,omitempty"`
+	ExecutorCoreRequest string                            `json:"executorCoreRequest,omitempty"`
+	ExecutorMemory      string                            `json:"executorMemory,omitempty"`
+	Status              NetworkPolicyRecommendationStatus `json:"status,omitempty"`
+}
+
+type NetworkPolicyRecommendationStatus struct {
+	State                    string `json:"state,omitempty"`
+	SparkApplication         string `json:"sparkApplication,omitempty"`
+	CompletedStages          int    `json:"completedStages,omitempty"`
+	TotalStages              int    `json:"totalStages,omitempty"`
```

Comment:
Would it be possible to add completion timestamps? (doesn't need to be done now, we can add them later)